### PR TITLE
fix: more changes for thirdparty rework

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

- Added multitenancy recipe with API and functions (without implementation)
- Updated Recipe and API implementations in TPEP and TPL
- Removed Config management functions from thirdparty recipes (since it's now in multitenancy recipe)
- Disabled tests for compilation **(ignore all test files for review)**

### What is already reviewed
- Thirdparty provider implementations (except Boxy)
- Recipe & API interfaces changes
- TypeProvider changes

### What needs to be reviewed
- Changes in ProviderInput and ProviderConfig models
- TPEP and TPL implementations
- Multitenancy Recipe and API interfaces
- Boxy Implementation

### What does not need review
- Ignore all test related files - *_test.go, testingUtils.go, etc

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [x] Update Boxy provider implementation to take the boxyURL input
- [x] Have a getTenantId function on the backend as well
- [x] Expose all the provider functions to the user so that they can build their own APIs for social login in case they don't want to use our middleware

### CODE REVIEW

File: recipe/thirdpartyemailpassword/main.go [hunk 1]
```diff
@@ -107,8 +107,8 @@ func SendEmailWithContext(input emaildelivery.EmailType, userContext supertokens
 	return (*instance.EmailDelivery.IngredientInterfaceImpl.SendEmail)(input, userContext)
 }
 
-func ThirdPartySignInUp(thirdPartyID string, thirdPartyUserID string, email string) (tpepmodels.SignInUpResponse, error) {
-	return ThirdPartySignInUpWithContext(thirdPartyID, thirdPartyUserID, email, &map[string]interface{}{})
+func ThirdPartyManuallyCreateOrUpdateUser(thirdPartyID string, thirdPartyUserID string, email string) (tpepmodels.ManuallyCreateOrUpdateUserResponse, error) {
+	return ThirdPartyManuallyCreateOrUpdateUserWithContext(thirdPartyID, thirdPartyUserID, email, &map[string]interface{}{})
```
Comment (rishabhpoddar) [inline recipe/thirdpartyemailpassword/main.go:111]: Need to add all new functions as well - like FetchThirdPartyConfigWithContext and all

File: recipe/thirdpartypasswordless/main.go [hunk 2]
```diff
@@ -30,12 +30,12 @@ func Init(config tplmodels.TypeInput) supertokens.Recipe {
...
-func ThirdPartySignInUpWithContext(thirdPartyID string, thirdPartyUserID string, email string, userContext supertokens.UserContext) (tplmodels.ThirdPartySignInUp, error) {
+func ThirdPartyManuallyCreateOrUpdateUserWithContext(thirdPartyID string, thirdPartyUserID string, email string, userContext supertokens.UserContext) (tplmodels.ManuallyCreateOrUpdateUserResponse, error) {
 	instance, err := getRecipeInstanceOrThrowError()
 	if err != nil {
-		return tplmodels.ThirdPartySignInUp{}, err
+		return tplmodels.ManuallyCreateOrUpdateUserResponse{}, err
 	}
-	return (*instance.RecipeImpl.ThirdPartySignInUp)(thirdPartyID, thirdPartyUserID, email, userContext)
+	return (*instance.RecipeImpl.ThirdPartyManuallyCreateOrUpdateUser)(thirdPartyID, thirdPartyUserID, email, userContext)
 }
```
Comment (rishabhpoddar) [inline recipe/thirdpartypasswordless/main.go:39]: Need to add new functions like FetchThirdPartyConfigWithContext

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/thirdpartypasswordless/main.go (hunk 2 above)
Comment (sattvikc) [reply recipe/thirdpartypasswordless/main.go:39]: This won't be required.

File: recipe/thirdpartyemailpassword/main.go (hunk 1 above)
Comment (sattvikc) [reply recipe/thirdpartyemailpassword/main.go:111]: Done

**Comment by sattvikc** [comment]
Created this - https://github.com/supertokens/supertokens-golang/pull/224. Closing this as cleaning up branches. There are 2 open comments here, but it is irrelevant now as the interfaces have changed.

**Pull request closed without merging**